cli/cmd/project/delete.go

```diff
@@ -33,8 +34,27 @@ func DeleteCmd(ch *cmdutil.Helper) *cobra.Command {
 				}
 			}
 
+			var deployedID string
+			rc, err := dotrillcloud.GetAll(path, ch.AdminURL)
+			if err != nil {
+				return err
+			}
+			if rc != nil {
+				deployedID = rc.ProjectID
+			}
+
 			if name == "" {
-				return fmt.Errorf("please provide a valid project name. Run `rill project list` to see the available projects")
+				if rc == nil {
+					return fmt.Errorf("please provide a valid project name. Run `rill project list` to see the available projects")
+				}
+
+				proj, err := client.GetProjectByID(cmd.Context(), &adminv1.GetProjectByIDRequest{
+					Id: deployedID,
+				})
+				if err != nil {
+					return err
+				}
+				name = proj.Project.Name
```

Comment:
This can be reverted because `GetProjectByID` is already called in `InferProjectName` above